### ISSUE TITLE
New Rule: don't extending same class

### ISSUE THREAD

**Issue by hvlxh** [eslint/eslint#17243]
### Rule details

don't extending same class. not gonna work

### Related ECMAScript feature

es6 and above

### What type of rule is this?

Warns about a potential problem

### Example code

```js
// wrong
class Client extends Client {

}

// correct
class Client {

}
```

### Why should this rule be in the core instead of a plugin?

i believe this one can be added in the core.

### Participation

- [X] I am willing to submit a pull request to implement this rule.

**Comment by Rec0iL99** [comment]
Hi @hvlxh, thanks for the issue. You can use the [`no-use-before-define`](https://eslint.org/docs/latest/rules/no-use-before-define) rule to catch class extending itself.

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdXNlLWJlZm9yZS1kZWZpbmU6IFwiZXJyb3JcIiovXG5cbmNsYXNzIENsaWVudCBleHRlbmRzIENsaWVudCB7XG5cbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)
